### PR TITLE
Showing options menu and navigation drawer using javascript & also getting nextTime1-4

### PR DESCRIPTION
**Pull request by infinyte7**
## Purpose / Description
Get nextTime1, nextTime2, nextTime3, nextTime4 in WebView & showing options menu and navigation drawer by javascript function called from WebView 
This is continuation #6307

## Fixes
Fixes _Link to the issues._

## Approach
_How does this change address the problem?_

## How Has This Been Tested?
Tested on android emulator
For showing Options Menu and Navigation Drawer using JavaScript
```
<button onclick="ankiShowOptionsMenu()">Show Options Menu</button>
<button onclick="ankiShowNavDrawer()">Show Navigation Drawer</button>
```

For getting time info
```
<script>
   console.log(AnkiDroidJS.ankiGetNextTime1());
   console.log(AnkiDroidJS.ankiGetNextTime2());
   console.log(AnkiDroidJS.ankiGetNextTime3());
   console.log(AnkiDroidJS.ankiGetNextTime4());
</script>
```

## Learning (optional, can help others)
https://developer.android.com/reference/android/webkit/WebView

## Checklist
_Please, go through these checks before submitting the PR._

- [ x ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ x ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ x ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ x ] You have commented your code, particularly in hard-to-understand areas
- [ x ] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 1]
```diff
@@ -3109,6 +3111,17 @@ private boolean filterUrl(String url) {
...
                 }
             }
+            // Show options menu through webview
+            if (url.startsWith("signal:show_options_menu")) {
+                openOptionsMenu();
+                return true;
+            }
+            // Show Navigation drawer menu through webview
+            if (url.startsWith("signal:show_navigation_drawer")) {
+                DrawerLayout mDrawerLayout = (DrawerLayout)findViewById(R.id.drawer_layout);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3121]: `super.onNavigationPressed()` should work.

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 2]
```diff
@@ -3399,5 +3412,25 @@ public boolean ankiGetCardMark() {
         public int ankiGetCardFlag() {
             return mCurrentCard.getUserFlag();
         }
+        
+        @JavascriptInterface
+        public String ankiGetNextTime1() {
+            return (String) mNext1.getText();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3418]: Can we get some sample values about what this returns (does it return the `<` for example)? We may also want to provide clients with the numeric value.

I'd rather avoid using encouraging people to use this value if at all possible, as they don't have the context to allow it to play well with text to speech "1d" should be read as "1 day".

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 3]
```diff
@@ -3109,6 +3111,17 @@ private boolean filterUrl(String url) {
                         return true;
                 }
             }
+            // Show options menu through webview
+            if (url.startsWith("signal:show_options_menu")) {
+                openOptionsMenu();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3116]: How does this work if the user is in fullscreen mode? Could you add a code comment here.

Comment (infinyte7) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3116]: Okay, I will add that

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java (hunk 2 above)
Comment (infinyte7) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3418]: So, for min - minute/minutes
 d - day/days, m - month/months, y - year/years

**Comment by infinyte7** [comment]
Again something messed up. I will clear it

**Comment by infinyte7** [comment]
Sorry for that, I am messing with git.

**Comment by david-allison-1** [comment]
@infinyte7 Don't worry! Feel free to start having a look through `git rebase`, as then you can force push to fix an incorrect branch

**Comment by mikehardy** [comment]
Yeah, my typical workflow here (assuming I am working on a single commit, as most of my PRs are just one commit)

1. git reset --soft HEAD~1  (this "un-commits" the last commit but keeps the changes local as if you had made the change but not committed)
1. do my edits in response to feedback
1. git commit -a (this commits the thing again but it has a new hash now)
1. git push origin +YOUR_BRANCH_NAME_HERE  (the '+' says "just overwrite the remote branch with this code - overwrite is potentially dangerous! be sure it is what you want)

That's basically it

To pull in upstream changes I have done this before
git remote add upstream git@github.com:ankidroid/Anki-Android.git

And periodically to get current:
1. git fetch upstream    (this syncs the local git cache with upstream changes but doesn't integrate them anywhere)
1. git rebase upstream/master  (this actually attempts to pull in changes from the 'upstream' remote's 'master' branch - it does that by stashing any change you have on the local branch that are not on the remote branch, adding in any new remote commits, then re-applying your commits

After you to a `git rebase upstream/master` your local commits will have different hashes then your personal remote branch, so you'll have to force-push again when you're ready

**Comment by infinyte7** [comment]
Thank you @mikehardy @david-allison-1 
I got back the github repo.
I will create PR in some time.

**Pull request closed without merging**